plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -648,4 +662,74 @@ protected TestView createSleepingView(Duration minimalSleepDuration)
+            return;
+        }
+        String tableName = "test_delete_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM region", 5);
+        assertUpdate("DELETE FROM " + tableName + " WHERE regionkey = 1", 1);
+        assertQuery("SELECT count(*) FROM " + tableName, "VALUES 4");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testMetadataDeleteWithVarcharEquality()
+    {
+        if (!hasBehavior(SUPPORTS_METADATA_DELETE) || !hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_EQUALITY)) {
+            assertQueryFails(
+                    "DELETE FROM nation WHERE name = 'ALBANIA'",
+                    "(.*[Dd]elet(e|ing).*(not |un)supported.*)" +
+                            "|(This connector does not support deletes)" +
+                            "|(Unsupported delete)");
+            return;
+        }
+        String tableName = "test_delete_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + "(col) AS VALUES 'a', 'A', null", 3);
+        assertUpdate("DELETE FROM " + tableName + " WHERE col = 'A'", 1);
+        assertQuery("SELECT * FROM " + tableName, "VALUES 'a', null");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testMetadataDeleteWithVarcharInequality()
+    {
+        if (!hasBehavior(SUPPORTS_METADATA_DELETE) || !hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_INEQUALITY)) {
+            assertQueryFails(
+                    "DELETE FROM nation WHERE name != 'ALBANIA'",
+                    "(.*[Dd]elet(e|ing).*(not |un)supported.*)" +
+                            "|(This connector does not support deletes)" +
+                            "|(Unsupported delete)");
+            return;
+        }
+        String tableName = "test_delete_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + "(col) AS VALUES 'a', 'A', 'b', null", 4);
+        assertUpdate("DELETE FROM " + tableName + " WHERE col != 'A'", 2);
+        assertQuery("SELECT * FROM " + tableName, "VALUES 'A', null");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testMetadataDeleteWithVarcharGreaterAndLower()
+    {
+        if (!hasBehavior(SUPPORTS_METADATA_DELETE) || !hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_INEQUALITY)) {
+            assertQueryFails(
+                    "DELETE FROM nation WHERE name > 'ALBANIA'",
+                    "(.*[Dd]elet(e|ing).*(not |un)supported.*)" +
+                            "|(This connector does not support deletes)" +
+                            "|(Unsupported delete)");
+            return;
+        }
+        String tableName = "test_delete_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + "(col) AS VALUES 'a', 'A', 'b', null", 4);
+        assertUpdate("DELETE FROM " + tableName + " WHERE col < 'A'", 0);
```

Comment:
> `, 0);` 

It tests that `A` < `a` but does not test delete.

Reply:
this is an empty delete. Below there is non empty.

Reply:
It's not possible to distinguish empty delete from non implemented delete.
```suggestion
        assertUpdate("DELETE FROM " + tableName + " WHERE col < 'a'", 1);
```

Reply:
It is. Non implemented will raise an error. While empty delete was executed in remote data source and it didn't remove anything.